webapp/packages/core-blocks/src/FormControls/InputField.tsx

```diff
@@ -141,12 +145,20 @@ export const InputField: InputFieldType = observer(
       value = mapState(value);
     }
 
-    const showRevealPasswordButton = rest.type === 'password' && !rest.readOnly;
-
-    if (showRevealPasswordButton && capsLock.warn) {
+    if (passwordType && !rest.readOnly && capsLock.warn) {
       description = translate('ui_capslock_on');
     }
 
+    useLayoutEffect(() => {
+      if (uncontrolled && value && inputRef.current) {
+        inputRef.current.value = value;
+      }
+    });
```

Comment:
`&& value` will be satisfied when the value is an empty string. It's not expected behavior. Please make strict checks for null or undefined based on the context.